automation/orchestrator.go

```diff
@@ -322,14 +322,24 @@ func (o *Orchestrator) RunWorkflow(ctx context.Context, wid workflow.ID, runID s
 	return runID, o.runWorkflow(ctx, wf, runID)
 }
 
+func (o *Orchestrator) lock() {
+	o.runLock.Lock()
```

Comment:
Could we get two log lines here, one before the call to `o.runLock.Lock()`, and one after? When the lock is already in use, that call will block, meaning it could be useful to see the time between them.